### PR TITLE
SPARKC-31: Added support for passing the limit clause to CQL in order to fetch top n results

### PR DESCRIPTION
**Pull request by jacek-lewandowski**
With this patch:
- added limit method to CassandraRDD, CassandraJavaRDD and CassandraJavaPairRDD
- updated CQL generation method in CassandraRDD so that it can use the limit clause
- added test case
- updated documentation and CHANGES.txt

### CODE REVIEW

File: doc/3_selection.md [hunk 1]
```diff
@@ -39,6 +39,17 @@ are currently allowed by the Cassandra engine. This limitation is going to be ad
...
+
+When a table is designed so that it include clustering keys and the use case is that only the first
+`n` rows from an explicitly specified Cassandra partition are supposed to be fetched, one can find
+useful the `limit` method. It allows to add the `LIMIT` clause to each CQL executed for a particular
+RDD.
+
+Note that, when you specify the limit without specifying a partition key predicate for the where
+clause, you will get unpredictable amount of rows because the limit will be applied on each Spark
+partition which is created for the RDD.
+
```
Comment (RussellSpitzer) [inline doc/3_selection.md]: Might want to add following this with .top or .takeordered (limit size) will enforce the limit on the Spark Side

File: doc/3_selection.md [hunk 2]
```diff
@@ -39,6 +39,17 @@ are currently allowed by the Cassandra engine. This limitation is going to be ad
 Cassandra releases. Currently, `ALLOW FILTERING` works well 
 with columns indexed by secondary indexes or clustering columns.  
 
+### Limiting rows
+
+When a table is designed so that it include clustering keys and the use case is that only the first
```
Comment (pkolaczk) [inline doc/3_selection.md]: it includes

File: doc/3_selection.md (hunk 1 above)
Comment (pkolaczk) [inline doc/3_selection.md]: Yeah, that would save some confusion, even though it would not be very useful.

**Comment by pkolaczk** [comment]
+1
Needs rebasing, but LGTM.

**Pull request merged**